feat: update release schedule information (#1941)
mendersoftware/mender-docs: 302.Release-information/01.Release-schedule/docs.md

---
title: Release versioning and schedule
taxonomy:
    category: docs
---

There are two Mender release types:

* LTS (Long Term Support)
* non-LTS (Standard Support)

Our quality requirements for both release types are the same; the only difference between LTS and non-LTS releases is the support length. We maintain LTS releases through patch releases for one year from the release date. In contrast, non-LTS releases get patch releases only if an urgent vulnerability is found between the release date and the next release.

Keeping these dates in mind helps decide which version to adopt and when to upgrade to a new version. In addition, ensure you are always using the latest available patch releases for the optimal operation of Mender.

##### Versioning

> 

Reviewer: ```suggestion
Mender is versioned and released as a Product bundle and the versions mentioned here refer to the Product bundle version.
Under the version of the Product bundle there are multiple different components with their individual versioning.
To see which component versions belong to a bundle please refer to [this json file](https://docs.mender.io/releases/versions.json)
```

@kacf 
This is how I had the rough version in mind.
Didn't get the whole autoversion part, sorry :(

> Mender users Semantic Versioning, and the version numbers are in the `MAJOR.MINOR.PATCH` format.

Reviewer 2: Wouldn't it be useful to link semver.org here?

We increment:

* `MAJOR` version when we make incompatible API changes;
* `MINOR` version when we add functionalities in a backward compatible manner;
* `PATCH` version when we make backward compatible bug fixes to existing releases.

!!! By design, the Mender Backend is always backward compatible with older versions of the Mender Client. Therefore, incrementing the `MAJOR` version means that the Mender Client (generally speaking, the client components) depend on a corresponding `MAJOR` version of the Management Server. However, any older Mender Client will still work with a Management Server running a more recent version.

#### Release Cadence

> We release new minor versions every three months, either as a non-LTS or LTS release. We publish LTS releases approximately every six months, but version numbers may vary depending on the release's content. Check back here to learn which versions are part of an LTS release series.

Reviewer: > but version numbers may vary depending on the release's content

Don't understand what this means (couldn't make up an example).

Author: next release is 3.5 or 4.0?

Reviewer: A ok, I would simplify that. 
The major/minor part has been covered already so no need to mention it here.

Also the times are usually not true, the 3 months is close enough, but using that to go further in the future is probably a false promise.

What do you think about the section below:

```suggestion
We release a new version every three months, either as a non-LTS or LTS release. As a rough rule, every second release is an LTS. Check back here to learn which versions are part of an LTS release series.
```

Reviewer 2: @TheMeaningfulEngineer the wording "rough rule" doesn't sit right with me. Its a very colloquial form (plus, the English saying is actually "rule of thumb"). My personal take is that the version by @tranchitella is more suitable.

Reviewer: Fair.

Though mentioning the minor version here complicates things (what if its a major?) and leaves the need for the unclear sentence.

What about this one:

```suggestion
We release a new minor/major version every three months, either as a non-LTS or LTS release. We publish LTS releases approximately every six months. Check back here to learn which versions are part of an LTS release series.
```

Author: Slightly modified sentence: We release a new version every three months, either as a non-LTS or LTS release. We publish LTS releases approximately every six months, but version numbers (e.g., major or minor version bumps) may vary depending on the release's content. Check back here to learn which versions are part of an LTS release series.

> We publish patch versions for older LTS releases roughly the same time we publish the new minor or major releases unless we discover an urgent vulnerability that requires an immediate fix.

Author: The "lot of delay" is a mistake, we aim at releasing them (almost) at the same time.
We don't want to document our failures.

> We publish patch versions for older LTS releases roughly the same time we publish the new minor or major releases unless we discover an urgent vulnerability that requires an immediate fix.

Reviewer: But the back-porting can turn into a significant effort (@kacf do correct me if I'm wrong) and we don't know that up front.
How about this:

```suggestion
The work to backport bugfixes to older LTS releases starts once the new minor or major release are published.
The release date depends on the complexity of the back-porting effort and can vary per release.
```

##### Current LTS releases

<!-- the version number includes the patch release here, to get picked up by autoversion, but will be a minor version in the result -->
<!--AUTOVERSION: "LTS releases: %"/lts -->
At this time, we support the following LTS releases: 3.3.

| LTS         | Supported until |
| ----------- | --------------- |
| 3.3         |  2023-06        |

##### Yocto LTS support

Our [meta-mender Yocto layer](https://github.com/mendersoftware/meta-mender) supports the latest Yocto LTS release, which at this time is Kirkstone (4.0).

> We aim at supporting the previous Yocto LTS release, which at this time is Dunfell (3.1) on a best effort basis.

Reviewer 3: IMHO, I don't think we should write this, we should improve our internal process instead. The agreement we already have from years ago is to support the two latest LTS releases. This just makes it ambiguous and difficult to understand.

Reviewer 2: Conceptually I agree with @kacf but having some form of clarifying information in until we have improved the process is beneficial IMHO.

Reviewer 3: I don't think that's clarifying, it's just confusing (why are we jumping back and forth).

@lluiscampos: What about putting a dunfell release in the release checklist as a stop-gap solution for now? At least then it won't slip through the cracks.

Reviewer: > I don't think that's clarifying, it's just confusing

Agree with @kacf here.
meta-mender Dunfell has been receiving the new recipes and we clarified what supported means.
I've been telling prospects it is supported (thought it was until recently)
And we discovered that best effort is a horrible word :D

If @lluiscampos also agrees on the suggestion on adding it to the checklist above, "supported" does seem to represent the real state versus the ambiguous best effort.

Author: We agreed that the assumption that we support only the latest release was wrong. I changed the paragraph stating that we support the latest two Yocto LTSes.

---

> title: Release versioning and schedule

Reviewer 4: Suggestion: `Release version and schedule`

taxonomy:
    category: docs
---

There are two Mender release types:

* LTS (Long Term Support)
* non-LTS (Standard Support)

> Our quality requirements for both release types are the same; the only difference between LTS and non-LTS releases is the support length. We maintain LTS releases through patch releases for one year from the release date. In contrast, non-LTS releases get patch releases only if an urgent vulnerability is found between the release date and the next release.

Reviewer 4: Nit: I really see no use for a `;` here. use `,`

> Keeping these dates in mind helps decide which version to adopt and when to upgrade to a new version. In addition, ensure you are always using the latest available patch releases for the optimal operation of Mender.

Author: replaced with "Keeping this distinction in mind".

##### Versioning

Mender users [Semantic Versioning](https://semver.org), and the version numbers are in the `MAJOR.MINOR.PATCH` format.

We increment:

> * `MAJOR` version when we make incompatible API changes;

Reviewer 4: Nit: I would drop the `;`
